cachi2/core/package_managers/rpm/redhat.py

```
@@ -0,0 +1,111 @@
+    lockfileVersion: PositiveInt
+    lockfileVendor: str
+
+
+class RedhatRpmsLock(RpmsLock):
+    def match_format(self) -> bool:
+        try:
+            header = Header(**self._content)  # parse lockfile header with pydantic
+        except ValidationError:
+            return False
+        # format specific match evaluation
+        return header.lockfileVendor == "redhat" and header.lockfileVersion == 1
+
+    def process_format(self) -> None:
+        # parse the whole lockfile with pydantic
+        self._lockfile = Root(**self._content)
+
+    def download(self, output_dir: RootedPath) -> None:
+        for arch in self._lockfile.arches:
+            log.debug(f"Downloading files for '{arch.arch}' architecture.")
+            files = {}  # files per URL for downloading packages & sources
+            for pkg in arch.packages:
+                dest = output_dir.join_within_root(arch.arch, pkg.repoid, os.path.basename(pkg.url))
+                files[pkg.url] = dest.path
+                self._files_metadata[dest.path] = {
+                    "package": True,
+                    "url": pkg.url,
+                    "size": pkg.size,
+                    "checksum": pkg.checksum,
+                }
+                os.makedirs(os.path.dirname(dest.path), exist_ok=True)
+
+            for pkg in arch.sources:
+                dest = output_dir.join_within_root(
+                    "sources", arch.arch, pkg.repoid, os.path.basename(pkg.url)
+                )
+                files[pkg.url] = dest.path
+                self._files_metadata[dest.path] = {
+                    "package": False,
+                    "url": pkg.url,
+                    "size": pkg.size,
+                    "checksum": pkg.checksum,
+                }
+                os.makedirs(os.path.dirname(dest.path), exist_ok=True)
+
+            asyncio.run(async_download_files(files, get_config().concurrency_limit))
+
+    def verify_downloaded(self) -> None:
```

Should I rather raise the exception and quit instead of displaying warnings in this method?